# Measures of Dispersion

Measures of dispersion are statistical methods used to understand how spread out or scattered the values are in a dataset. While central tendency tells us the center of the data, dispersion tells us how much the values differ from each other and from the center.

## Range
Range is the difference between the highest and lowest values in a dataset. It gives a simple idea of how spread out the data is.

## Formula
Range=Maximum−Minimum

## Numerical Example
Marks = [45, 50, 55, 60, 65, 70, 75, 80, 85, 90]

Maximum = 90
Minimum = 45

Range = 90 - 45 = 45

In [1]:
## Python Implementation
marks = [45, 50, 55, 60, 65, 70, 75, 80, 85, 90]

print(max(marks) - min(marks))

45


Here, max() finds the highest mark, 90, and min() finds the lowest mark, 45. We subtract 45 from 90, so the range is 45.

## Variance
Variance measures how far the values in a dataset are spread out from their mean. Higher variance means the values are more spread out.

## Formula

Variance = Σ(X - μ)² / N

Where:

X = Individual value

μ = Population mean

N = Number of observations

The differences are squared so that negative and positive differences do not cancel each other.

## Numerical Example
Marks = [40, 50, 60]

Mean = 50

(40 - 50)² = 100
(50 - 50)² = 0
(60 - 50)² = 100

Variance = (100 + 0 + 100) / 3
Variance = 66.67

In [2]:
## Python Implementation
import statistics

marks = [40, 50, 60]

variance = statistics.pvariance(marks)

print(variance)

66.66666666666667


Here, the mean of the marks is 50. Python calculates the difference between each mark and the mean, squares the differences, and finds their average. The resulting variance is 66.67.

## Standard Deviation
Standard deviation measures how much the values in a dataset typically differ from the mean. A smaller standard deviation means the values are closer to the mean, while a larger one means the values are more spread out.

### Formula
Formula Standard Deviation = √Variance

For population data:

σ = √[Σ(X - μ)² / N]

### Numerical Example
Marks = [40, 50, 60]

Mean = 50

Variance = 66.67

Standard Deviation = √66.67
Standard Deviation ≈ 8.16

In [3]:
### Python Implementation
import statistics

marks = [40, 50, 60]

std = statistics.pstdev(marks)

print("Standard Deviation:", std)

Standard Deviation: 8.16496580927726


### AI/ML Engineer Example
An ML engineer can use standard deviation to understand how much a feature such as customer age, salary, or transaction amount varies from its average. It can also help identify unusual values in the dataset.

## Quartiles
Quartiles divide an ordered dataset into four approximately equal parts. The three main quartiles are Q1, Q2, and Q3, representing the 25th, 50th, and 75th percentiles.

### Formula
Q1 = 25th Percentile
Q2 = 50th Percentile
Q3 = 75th Percentile

### Numerical Example
Marks = [10, 20, 30, 40, 50, 60, 70]

Q1 = 25
Q2 = 40
Q3 = 55

### AI/ML Engineer Example
An ML engineer can use quartiles to understand the distribution of features such as customer spending or income. Quartiles can also help identify unusual values and understand how the data is distributed.

In [4]:
### Python Implementation
import numpy as np

marks = [10, 20, 30, 40, 50, 60, 70]

Q1 = np.percentile(marks, 25)
Q2 = np.percentile(marks, 50)
Q3 = np.percentile(marks, 75)

print("Q1:", Q1)
print("Q2:", Q2)
print("Q3:", Q3)

Q1: 25.0
Q2: 40.0
Q3: 55.0


Here, np.percentile() is used to calculate the three quartiles. We use 25 for Q1, 50 for Q2, and 75 for Q3. The results are 25, 40, and 55, which help us understand different positions within the dataset.

## Percentile
A percentile shows the relative position of a value within a dataset. It tells us the value below which a particular percentage of observations fall.

### AI/ML Engineer Example

An ML engineer can use percentiles to identify the top or bottom portion of data, such as customers in the top 10% of spending. Percentiles can also be used to create thresholds for detecting unusual values.

In [5]:
#Python Implementation
import numpy as np

marks = [45, 50, 55, 60, 65, 70, 75, 80, 85, 90]

percentile_75 = np.percentile(marks, 75)

print("75th Percentile:", percentile_75)

75th Percentile: 78.75


Here, np.percentile() calculates the value at the 75th percentile. The result is 78.75, meaning approximately 75% of the observations are at or below this value.

## Interquartile Range (IQR)

Interquartile Range (IQR) measures the spread of the middle 50% of the data. It is useful for understanding data spread and detecting outliers because it is less affected by extreme values.

###Formula
IQR=Q3−Q1

### Numerical Example
Q1 = 25
Q3 = 55

IQR = Q3 - Q1

IQR = 55 - 25

IQR = 30

### AI/ML Engineer Example

An ML engineer can use IQR to detect outliers in features such as transaction amount, salary, or delivery time. Values significantly below Q1 or above Q3 can be investigated as potential outliers.

In [6]:
### Python Implementation
import numpy as np

marks = [10, 20, 30, 40, 50, 60, 70]

Q1 = np.percentile(marks, 25)
Q3 = np.percentile(marks, 75)

IQR = Q3 - Q1

print("Q1:", Q1)
print("Q3:", Q3)
print("IQR:", IQR)

Q1: 25.0
Q3: 55.0
IQR: 30.0


Here, we first calculate Q1 as 25 and Q3 as 55. We then subtract Q1 from Q3 using the IQR formula. The result is 30, meaning the middle 50% of the data is spread across 30 units.

## Coefficient of Variation

Coefficient of Variation (CV) measures the amount of variation in a dataset relative to its mean. It is expressed as a percentage and is useful for comparing the consistency of datasets with different scales.

### Formula

CV = (Standard Deviation / Mean) × 100

A lower CV means less variation relative to the mean.

A higher CV means more variation relative to the mean.

### Numerical Example
Mean = 50
Standard Deviation = 5

CV = (5 / 50) × 100

CV = 10%

### AI/ML Engineer Example

An ML engineer can use CV to compare the relative variability of different features, even when their values have different scales. A lower CV generally indicates that a feature is more consistent, while a higher CV indicates greater variation.

In [7]:
#Python Implementation

import statistics

marks = [45, 48, 50, 52, 55]

mean = statistics.mean(marks)
std = statistics.pstdev(marks)

cv = (std / mean) * 100

print("Mean:", mean)
print("Standard Deviation:", round(std, 2))
print("Coefficient of Variation:", round(cv, 2), "%")

Mean: 50
Standard Deviation: 3.41
Coefficient of Variation: 6.81 %


Here, we first calculate the mean, which is 50, and the standard deviation, which is approximately 3.41. We divide the standard deviation by the mean and multiply by 100. The resulting CV is approximately 6.81%, which represents the variation relative to the average value.